script/main_test.py

```diff
@@ -1,5 +1,24 @@
 import unittest
 import main
+import sqlite3
+
+
+def create_connection(db_file):
+  conn = None
+  try:
+    conn = sqlite3.connect(db_file)
+  except sqlite3.Error as e:
+    print(e)
+  
+  return conn
+
+
+class TestMostPopularCities(unittest.TestCase):
+
+  def test_most_popular_cities(self):
+    self.assertEqual(len(main.most_popular_cities(create_connection("../db/pythonsqliteusers.db"), 5)), 5)
```

Comment:
Create connection once and then use it.
also use setUp method here.
https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp

Reply:
I don't know what I'm doing wrong...

```
class TestMostPopularCities(unittest.TestCase):

  def setUp(self):
    self.connection = create_connection("../db/pythonsqliteusers.db")

  def test_most_popular_cities(self):
    self.assertEqual(len(main.most_popular_cities(connection, 5)), 5)
```

and I get an error: 
`NameError: name 'connection' is not defined`

Reply:
I solved it ... I forgot the word "self"